nomad/encrypter.go

```diff
@@ -457,6 +456,7 @@ START:
 				goto ERR_WAIT // rate limit exceeded
 			}
 
+			store := krr.srv.fsm.State()
 			ws := store.NewWatchSet()
```

Comment:
I'm going to admit I feel like I'm supposed to be able to have this `store.NewWatchSet` at the top of the loop and then I can `select` on it with the rest of the context to replace the state handle... but I can't figure out which is the right thing to be polling on there and this diff fixes the test case I've got.

Reply:
So I think this can restructured a bit, but I could be wrong as I'm new to this area of code:

`(*state.StateStore).AbandonCh()` will get closed when a restore finishes, so I think just toss it in the select above and then you only need to do `store = krr.srv.fsm.State()` in that case block.

The `WatchSet` can be dropped and `nil` apssed t`WatchSets` are internal to iradix indexes while Abandoning is a StateStore concept. I think this is an accurate mental model:

```
Raft
 |
 v
FSM   <readers>
 |      ^
 v      |
StateStore <- Abandon
  |
  v
MemDB
  |
  v
iradix indexes <- Watches
```

If replicators were waiting for _their local root key meta to be updated via the FSM_ then I think the watchset could be added to the `select {}` to be notified when it was updated. Since the root key meta is replicated by RPC outside of the FSM, I don't think a watchset is useful here.

But like I said: new code to me, so let's maybe zoom if this all sounds way off.